File: app/src/main/java/com/tuvakov/zetube/android/repository/SubscriptionRepo.java
package com.tuvakov.zetube.android.repository;

import androidx.lifecycle.LiveData;

import com.tuvakov.zetube.android.data.Subscription;
import com.tuvakov.zetube.android.data.SubscriptionDao;

import java.util.List;
import java.util.concurrent.ExecutorService;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class SubscriptionRepo {

    private final SubscriptionDao mSubscriptionDao;
    private final ExecutorService mDiskIO;
    private LiveData<List<Subscription>> mAllSubscriptions;

    @Inject
    public SubscriptionRepo(SubscriptionDao subscriptionDao, ExecutorService diskIO) {
        mSubscriptionDao = subscriptionDao;
        mDiskIO = diskIO;
        mAllSubscriptions = mSubscriptionDao.selectAll();
    }

    public void insert(Subscription subscription) {
        mDiskIO.execute(() -> mSubscriptionDao.insert(subscription));
    }

    public void bulkInsertForService(List<Subscription> subscriptions) {
        mSubscriptionDao.bulkInsert(subscriptions);
    }

    public void update(Subscription subscription) {
        mDiskIO.execute(() -> mSubscriptionDao.update(subscription));
    }

    public void delete(Subscription subscription) {
        mDiskIO.execute(() -> mSubscriptionDao.delete(subscription));
    }

    public void deleteAll() {
        mDiskIO.execute(mSubscriptionDao::deleteAll);
    }

    public void deleteAllForService() {
        mSubscriptionDao.deleteAll();
    }

    public LiveData<List<Subscription>> getAllSubscriptions() {
        return mAllSubscriptions;
    }
}

File: app/src/main/java/com/tuvakov/zetube/android/repository/SubscriptionRepo.kt
package com.tuvakov.zetube.android.repository

import androidx.lifecycle.LiveData
import com.tuvakov.zetube.android.data.Subscription
import com.tuvakov.zetube.android.data.SubscriptionDao
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class SubscriptionRepo @Inject constructor(private val mSubscriptionDao: SubscriptionDao) {

    val subscriptions: LiveData<List<Subscription>> = mSubscriptionDao.subscriptions

    suspend fun getSubscriptionById(id: String) = mSubscriptionDao.getSubscriptionById(id)

    suspend fun insert(subscription: Subscription) {
        mSubscriptionDao.insert(subscription)
    }

    suspend fun bulkInsert(subscriptions: List<Subscription>) {
        mSubscriptionDao.bulkInsert(subscriptions)
    }

    suspend fun update(subscription: Subscription) {
        mSubscriptionDao.update(subscription)
    }

    suspend fun delete(subscription: Subscription) {
        mSubscriptionDao.delete(subscription)
    }

    suspend fun deleteAll() {
        mSubscriptionDao.deleteAll()
    }
}